runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -672,11 +706,84 @@ public void freeTableData(String namespace, String tableName) {
      * @param namespace Namespace of the table.
      * @param tableName Name of the table.
      */
-    public void deleteTable(String namespace, String tableName) {
+    public void clearTable(String namespace, String tableName) {
         Table<Message, Message, Message> table = getTable(namespace, tableName);
         table.clearAll();
     }
 
+    /**
+     * Clear the table by appending CLEAR to the stream. Start a transaction to mark the
+     * table as dropped in RegistryTable, and add this table and its checkpoint stream to the
+     * transaction's deleteSet, which instructs the sequencer to clean up the in-mem maps.
+     *
+     * @param namespace Namespace of the table.
+     * @param tableName Name of the table.
+     */
+    public void dropTable(String namespace, String tableName) {
+        log.info("Invoking dropTable {}${}", namespace, tableName);
+        clearAndUnregisterTable(namespace, tableName);
+
+        String fullyQualifiedTableName = getFullyQualifiedTableName(namespace, tableName);
+        tableMap.remove(fullyQualifiedTableName);
+        log.info("Done dropping table {}${}", namespace, tableName);
+    }
+
+    private void clearAndUnregisterTable(String namespace, String tableName) {
+        TableName tableNameKey = TableName.newBuilder()
+                .setNamespace(namespace)
+                .setTableName(tableName)
+                .build();
+
+        // Mark this table as dropped in RegistryTable
+        int numRetries = 9; // Since this is an internal transaction, retry a few times before giving up.
+        while (numRetries-- > 0) {
+            if (TransactionalContext.isInTransaction()) {
+                throw new IllegalThreadStateException("openTable: Called on an existing transaction");
+            }
+            try {
+                this.runtime.getObjectsView().TXBuild()
+                        .type(TransactionType.WRITE_AFTER_WRITE)
+                        .build()
+                        .begin();
+
+                // Clear table data
+                UUID streamId = CorfuRuntime.getStreamID(getFullyQualifiedTableName(namespace, tableName));
+                SMREntry entry = new SMREntry("clear", new Array[0], Serializers.PRIMITIVE);
+                TransactionalContext.getCurrentContext().logUpdate(streamId, entry);
```

Comment:
Do you write a raw SMR entry to prevent opening the table here?

Isn't clear a pure mutator and therefore won't sync the table?

Reply:
Yes. Not opening the table makes this API more versatile (for example, dropping a corrupted table that can't be opened).